src/Calculator/Controls/EquationTextBox.cpp

```diff
@@ -50,13 +56,18 @@ void EquationTextBox::OnApplyTemplate()
         m_equationButton->Click += ref new RoutedEventHandler(this, &EquationTextBox::OnEquationButtonClicked);
 
          auto toolTip = ref new ToolTip();
-         auto resProvider = AppResourceProvider::GetInstance();
          auto equationButtonMessage = m_equationButton->IsChecked->Value ? resProvider->GetResourceString(L"showEquationButtonToolTip") : resProvider->GetResourceString(L"hideEquationButtonToolTip");
          toolTip->Content = equationButtonMessage;
          ToolTipService::SetToolTip(m_equationButton, toolTip);
          AutomationProperties::SetName(m_equationButton, equationButtonMessage);
     }
 
+    if (m_richEditContextMenu != nullptr)
+    {
+        m_richEditContextMenu->Opening +=
```

Comment:
Visual state bug:
1. Type "y=mx + b" but don't submit
2. Right click text box and click analyze equation
3. Click back button on KGF page
4. Equation text box is in the focus visual state even though it does not have focus

Reply:
I spoke with Rudy about this, and he has a separate fix for this that manages focus on transition to and from KGF.